test/optional-validation.test.js

```diff
@@ -133,6 +133,24 @@ describe('optional-validation', function() {
           });
         });
       });
+
+      it('should fail if the `Primary Key` is set for `instance`' +
+      'in `before save` operation hook', function(done) {
+        function changeIdInBeforeSaveHook() {
+          ModelWithForceId.observe('before save', function(ctx, next) {
+            ctx.instance.id = 50;
+            next();
+          });
+        }
+        ModelWithForceId.create({ name: 'foo' }, function(err, created) {
+          if (err) return done(err);
+          changeIdInBeforeSaveHook();
+          created.replaceAttributes({ name: 'bar' }, function(err, data) {
+            should.exist(err);
```

Comment:
Please verify that the you received the expected error, and not some other error instead.

For example:

```js
err.message.should.match(/id property cannot be changed.*forceId/);
```